### ISSUE TITLE
[Important] The current website is about to enter maintenance mode to aid with v3 launch

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/c#559]
**TL;DR;** At the end of Jan 2021, all tracks will enter v3 staging mode. Updates will no longer sync with the current live website, but instead sync with the [staging website](https://exercism.lol/). The C section of the v3 repo will be extracted and PR'd into this track (if appropriate). Further issues and information will follow over the coming weeks to prepare C for the launch of v3.

Over the last 12 months, we've all been hard at work developing [Exercism v3](https://github.com/exercism/v3#exercism---v3). Up until this point, all v3 tracks have been under development in a single repository - the [v3 repository](https://github.com/exercism/v3). As we get close to launch, it is time for us to explode that monorepo back into the normal track repos. Therefore, at the end of this month (January 2021), we will copy the v3 tracks contents from the v3 repository back to the corresponding track repositories.

As v3 tracks are structured differently than v2 tracks, the current (v2) website cannot work with v3 tracks. To prevent the v2 website from breaking, we'll disable syncing between track repositories and the website. This will effectively put v2 in maintenance mode, where any changes in the track repos won't show up on the website. This will then allow tracks to work on preparing for the Exercism v3 launch.

Where possible, we will script the changes needed to prepare tracks for v3. For any manual changes that need to be happening, we will create issues on the corresponding track repositories. We will be providing lots of extra information about this in the coming weeks.

We're really excited to enter the next phase of building Exercism v3, and to finally get it launched! 🙂

**Comment by ErikSchierboom** [comment]
As of today, the website has entered maintenance mode.

**Comment by guygastineau** [comment]
I am a little confused (which is my fault for drifting away from exercism during this v3 saga).  v3 seems cool, but I just popped by, because I would like to port the `Forth` exercise to C for fun.  Am I understanding correctly, that if I port `Forth` to the C track it won't actually get synced?

**Comment by wolf99** [comment]
Correct, you can add exercises but they won't show up on the v2 website. They will show up on v3 website and be available for completion when that site moves out of development.

If you're interested to add exercises in general, we could do with some concept exercises as we have none yet!
The concepts are already listed though 😊, see https://github.com/exercism/c/tree/main/reference and the docs within that directory.

**Comment by guygastineau** [comment]
Thanks for the explanation.  I figured that was basically the case once I started porting forth and realized that the exercises were in `exercises/practice` now.

BTW, I think I developed a sensible interface for Forth.  I might get a WIP PR up tomorrow, but I should catch up on some work too, so that might not happen.

Side note about that, I couldn't figure out how to have multiple modules in an example implementation, but my VM based solution to forth would really benefit from isolating the dynamic arrays involved in the different parts.

**Comment by wolf99** [comment]
How to have multiple modules is an interesting question!

It's not something any exercise we've implemented to date has needed.

Additionally, I don't know if the exercise `config.json` would allow this. But I think that it would be good if it did not specifically prevent this if it is not intended. I will seek info on the idea on the Slack.

**Comment by guygastineau** [comment]
Thanks for looking into it.  It felt a little off topic for this issue thread, but I am in a hurry on my phone, so I am being lazy

**Comment by wolf99** [comment]
So the fields in the exercise config.json are arrays, which means it's possible to add multiple files.
But it's currently not supported for a student to create an arbitrary file..

Off the top of my head I would have thought it possible to implement this without needing multiple modules... but I haven't thought it through in any detail.

Anyway, if you have ideas, put up a PR or issue and we can discuss further there 😃

**Comment by guygastineau** [comment]
Sure. I obviously didn't get it up this weekend, but by the end of the week I will put my work up in a PR.

Certainly, I don't need to use multiple modules, but a non-naive solution is made much nicer by having them.  For instance, implementing non-null-terminated string slices (something like)
``` c
struct string_slice {
  size_t length;
  const char *text
};

int ss_cmp(struct string_slice x, struct string_slice y);

// Some other helpers
// ...
```
helps keep from copying strings too much when tokenizing and using the symbol table for Forth variables.  It is not necessary in this toy implementation, but if any real program were interpreted by our Forth implementation I think the memory performance would show.

Similarly, I am writing a hashtable implementation for the symbol table.  A simple association list or dynamic array of key value pairs would suffice (although those implementations would also benefit from being in their own modules), but once again, at scale these data structures would obviously introduce too much time complexity for operations that should be very cheap.

Anyway, we can hold off on getting into the meat of that discussion until I have a PR up.  I am feeling positive that we can do multiple modules though.  the `config.json` shouldn't matter too much, so long as we can add files to it.  The example solution doesn't matter much to the config.json I thought.  What I know needs to be able to handle multiple modules is the test building system used for the CI in this track.  I am willing to help refactor that for this purpose too, but I am also willing to jam everything into one header for this example exercise as well.
